### PR TITLE
Add information icon to checkboxes in Settings window

### PR DESCRIPTION
**Pull request by DmitryZhelnin**
## Proposed changes
Some of the checkboxes in the Settings window have additional information that is hidden inside tooltip. For the user it is not so obvious that some of them have tooltips.
This PR adds new UserControl named `SettingsCheckBox`. Control has info icon near the checkbox if it has a tooltip.

### Before

изображение

### After

изображение

## Test methodology 

Manual testing.

## Test environment(s) 

- GIT 2.23
- Windows 7 SP1

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by mstv** [comment]
Good idea. I have no strong opinion on the approach used nor had a deeper look into it.
Though there are plans to redesign the dialog using `PropertyGrid`.

**Comment by RussKie** [comment]
Please squash and we take it.

**Comment by DmitryZhelnin** [comment]
> 
> 
> Please squash and we take it.

@RussKie  Done

File: GitUI/CommandsDialogs/SettingsDialog/Pages/AdvancedSettingsPage.Designer.cs [hunk 1]
```diff
@@ -310,8 +308,7 @@ private void InitializeComponent()
             this.chkConsoleEmulator.Size = new System.Drawing.Size(1572, 17);
             this.chkConsoleEmulator.TabIndex = 3;
             this.chkConsoleEmulator.Text = "Use Console Emulator for console output in command dialogs";
-            this.tooltip.SetToolTip(this.chkConsoleEmulator, resources.GetString("chkConsoleEmulator.ToolTip"));
-            this.chkConsoleEmulator.UseVisualStyleBackColor = true;
```
Comment (mstv) [inline GitUI/CommandsDialogs/SettingsDialog/Pages/AdvancedSettingsPage.Designer.cs:314]: Does this still work well with the dark theme?

Comment (DmitryZhelnin) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/AdvancedSettingsPage.Designer.cs:314]: Yes, I checked dark theme and it looks similar

Comment (DmitryZhelnin) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/AdvancedSettingsPage.Designer.cs:314]: изображение

Comment (pmiossec) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/AdvancedSettingsPage.Designer.cs:314]: > Yes, I checked dark theme and it looks similar

Except the selection in the tree as we could see in the screenshot (text unreadable). But indeed, that's not your fault... /cc @NikolayXHD ?

Comment (RussKie) [reply GitUI/CommandsDialogs/SettingsDialog/Pages/AdvancedSettingsPage.Designer.cs:314]: > Does this still work well with the dark theme?

Thank you for asking! I need to learn to ask about the dark theme :)

**Comment by RussKie** [comment]
I'm taking this into v3.4 as well

**Pull request merged**